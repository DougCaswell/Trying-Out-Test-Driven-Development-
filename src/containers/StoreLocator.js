import React, { Component } from 'react';
import Header from '../components/Header';
import Button from '../components/Button';
import Map from '../components/Map';
import mapChooser from '../mapChooser';
import Axios from 'axios';

class StoreLocator extends Component {
    constructor(props) {
        super(props);

        this.state = {
            currentMap: 'none.png',
            shops: []
        }

        this.chooseMap = this.chooseMap.bind(this);
    }

    async componentDidMount() {
        let response = await Axios.get('http://localhost:3000/data/shops.json');
        console.log(response)
        this.setState({
            shops: response.data.shops
        })
    }

    chooseMap(e) {
        this.setState({ currentMap: mapChooser(e.target.value) });
    }

    render() {
        console.log(this.state.shops)

        let storeButtons = this.state.shops.map((shop, id) => {
            return (<Button handleClick={this.chooseMap} location={shop.location} key={id} />);
        });

        return (
            <div className='StoreLocator'>
                <Header />
                <div className='buttonContainer'>
                    {storeButtons}
                </div>
                <Map imageName={this.state.currentMap} />
            </div>
        );
    }
}

export default StoreLocator;