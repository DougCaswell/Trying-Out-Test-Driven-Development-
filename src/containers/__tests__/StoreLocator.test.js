import React from 'react';
import { shallow } from 'enzyme';
import StoreLocator from '../StoreLocator';
import axios from 'axios';

describe("StoreLocator", function () {

    let mountedStoreLocator;

    beforeEach(() => {
        mountedStoreLocator = shallow(<StoreLocator />);
    })

    it('calls axios.get in #componentDidMount', () => {
        return mountedStoreLocator.instance().componentDidMount().then(() => {
            expect(axios.get).toHaveBeenCalled();
        });
    })

    it('calls axios.get with correct url', () => {
        return mountedStoreLocator.instance().componentDidMount().then(() => {
            expect(axios.get).toHaveBeenCalledWith('http://localhost:3000/data/shops.json');
        });
    })

    it('updates state with api data', () => {
        return mountedStoreLocator.instance().componentDidMount().then(() => {
            expect(mountedStoreLocator.state()).toHaveProperty('shops', [
                {
                    "location": "test location",
                    "address": "123 Portland Dr"
                }
            ]);
        });
    })

    it('Shallow rendering of StoreLocator should work', () => {
        let mountedStoreLocator = shallow(<StoreLocator />);
    });

    it('renders a header', () => {
        const headers = mountedStoreLocator.find('Header');
        expect(headers.length).toBe(1);
    });

    // it('renders three buttons', () => {
    //     const buttons = mountedStoreLocator.find('Button');
    //     expect(buttons.length).toBe(3);
    // })

    it('renders a map', () => {
        const maps = mountedStoreLocator.find('Map');
        expect(maps.length).toBe(1);
    })

})

describe('chooseMap', () => {
    it('updates this.state.currentMap using the loaction passed to it', () => {
        let mountedStoreLocator = shallow(<StoreLocator />);
        let mockEvent = { target: { value: 'testland' } }
        mountedStoreLocator.instance().chooseMap(mockEvent);
        expect(mountedStoreLocator.instance().state.currentMap).toBe('testland.png')
    })
})