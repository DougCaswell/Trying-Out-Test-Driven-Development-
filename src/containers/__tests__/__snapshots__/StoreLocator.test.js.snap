// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator renders correctly 1`] = `
<div
  className="StoreLocator"
>
  <div
    className="Header"
  >
    <img
      alt="Wired Brain"
      src="images/wired-brain-coffee-logo.png"
    />
  </div>
  <div
    className="buttons"
  />
  <div
    className="MapBox"
  >
    <img
      alt="Map of all shops"
      src="images/none.png"
    />
  </div>
</div>
`;
